operations/helm/charts/mimir-distributed/values.yaml

```
@@ -3517,3 +3739,9 @@ continuous_test:
 #   data: |
 #     extra.yml: "does-my-install-need-extra-info: true"
 extraObjects: []
+
+# -- Add a set of dedicated query-frontend and queriers for the ruler to use.
+# -- This is useful for isolating the ruler queries from other queriers (api/grafana).
+# https://grafana.com/docs/mimir/latest/operators-guide/architecture/components/ruler/#remote
+ruler_querier_service:
+  enabled: false
```

```suggestion
remote_rule_evaluation:
  enabled: false
```

I think this name better aligns with the terminology we use elsewhere, WDYT?

Obviously, we'd need to update all the references in this PR as well.

---

Is it possible to wire this in when the ruler-querier replicas are more than 0?

---

With this feature we would have 2 evaluation modes but with 2 implementations of remote_rule_evaluation (shared vs dedicated ruler query paths)   We should probably distinguish those in the name ? Unless we simply have remote evaluation always deploy a dedicated query path ?  @Logiraptor @dimitarvdimitrov

---

> Is it possible to wire this in when the ruler-querier replicas are more than 0?

@dimitarvdimitrov  the default values for ruler-querier will always have the replicas > 0

---

I see. They have > 0 replicas, but are sometimes not created.

---

But the operator can choose to set the dedicated rulers to 0 replicas and use the shared query path. This will give them e.g. query sharding and query scheduling, which  the ruler itself doesn't support; all without having to deploy more resources.

I suggest to make it possible to deploy in both ways - dedicated and shared. Dedicated when the dedicated query-frontend replicas are more than 0 and shared when they are 0. 

If you agree, then I'd also suggest to make the dedicated ruler path deploy with 0 replicas. The reason is that we aim to have the chart more compatible with smaller deployments, but still enable larger deployments. A dedicated ruler path is a somewhat advanced feature and many regular operators might not need. Once we do that it would be great to document how to setup the remote ruler evaluation both with dedicated query path and a shared query path.

---

thoughts on this @gonzalez?